fn/concurrency.go

```diff
@@ -32,3 +34,43 @@ func ParSlice[V any](ctx context.Context, s []V, f ErrFunc[V]) error {
 
 	return errGroup.Wait()
 }
+
+// ParSliceErrCollect can be used to execute a function on each element of a
+// slice in parallel. This function is fully blocking and will wait for all
+// goroutines to finish. Error will be collected and returned as a map of slice
+// element index to error. Active goroutines limited with number of CPU.
+func ParSliceErrCollect[V any](ctx context.Context, s []V,
+	f ErrFunc[V]) (map[int]error, error) {
+
+	errGroup, ctx := errgroup.WithContext(ctx)
+	errGroup.SetLimit(runtime.NumCPU())
+
+	var instanceErrorsMutex sync.Mutex
+	instanceErrors := make(map[int]error, len(s))
```

Comment:
nit: can use `lnutils.SyncMap` here. Though fetching only the keys would be more ergonomic with what you have now IMO.

Reply:
I figured I could return `map[int]error` directly rather than use `lnutils.SyncMap`. And that the sync handling mutex would be restricted to the scope of this small function anyway.

That's why I went with a plain map over anything else. Any thoughts on my reasoning here?